src/ui/actions/logpoint.ts

```diff
@@ -262,7 +243,13 @@ async function setMultiSourceLogpoint(
       if (!points.error) {
         showPrimitiveLogpoints(logGroupId, points.data || [], primitiveFronts);
       }
-      return;
+      // If we're only displaying only primitives, we can bail out if there's no condition or the condition request is done
+      if (
+        !condition ||
+        (points.status === AnalysisStatus.Completed && points.condition === condition)
+      ) {
+        return;
```

Comment:
In the case that we have not seen this combo of locatoin & condition before, I would expect that `points` would not exist (because `getAnalysisPointsForLocation(store.getState(), locations[0], condition)` would return `undefined` and thus we would never get into this block in the first place because of the check on line 246. What am I missing?

Reply:
This was specifically for later hits to this block.  In particular:

- Add a print statement. Initial message is just primitives, like `"someFunc", 123`.  Don't have to request an analysis.
- Add a condition.  We need to request an analysis.
- Change the condition. The latest analysis _did_ complete and has points.... _but_ the condition changed, so we need to run a _new_ analysis.  What I was seeing was that prior to adding the "if condition changed" check, we were bailing out early because the last analysis we ran did succeed, it was just for a condition that was now different.

Reply:
So `getAnalysisPointsForLocation(store.getState(), locations[0], condition)` ignores the condition and just returns the most recently run analysis points?